src/components/Dropdown.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React, { useMemo, useState } from 'react';
+import cn from 'classnames';
+import { Person } from '../types/Person';
+
+interface Props {
+  persons: Person[];
+  onSelect: (person: Person) => void;
+}
+
+export const Dropdown: React.FC<Props> = ({ persons, onSelect }) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState(query);
+  const [isActiveSuggestions, setIsActiveSuggestions] = useState(false);
+
+  const handlePersonSelect = (person: Person) => {
+    setQuery(person.name);
+    setAppliedQuery(person.name);
+    onSelect(person);
+  };
+
+  const handleQuery = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    delay: number,
+  ) => {
+    if (isActiveSuggestions) {
+      setIsActiveSuggestions(false);
+    }
+
+    setQuery(event.target.value);
+
+    setTimeout(() => {
+      setAppliedQuery(event.target.value);
+      setIsActiveSuggestions(true);
+    }, delay);
```

Comment:
It will be triggered every time when user is typing something. Should  use separate debounce function